### PR TITLE
Cross build for Scala.js

### PR DESCRIPTION
**Pull request by armanbilge**
Towards https://github.com/http4s/http4s/pull/4938.

This was a pure cross (no changes necessary) besides for the file-based test suite.

### CODE REVIEW

**Review by ChristopherDavenport — APPROVED** [review]

File: build.sbt [hunk 1]
```diff
@@ -9,27 +9,36 @@ ThisBuild / publishFullName := "Ross A. Baker"
...
-versionIntroduced := Map(
-  "3.0.0-M1" -> "1.0.1",
-  "3.0.0-M2" -> "1.0.1",
-  "3.0.0-M3" -> "1.0.1",
-  "3.0.0-RC1" -> "1.1.0",
-  "3.0.0-RC2" -> "1.1.1",
-  "3.0.0-RC3" -> "1.1.2"
-)
+lazy val `jawn-fs2` = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Full)
```
Comment (mpilquist) [inline build.sbt:23]: Is this needed?

File: build.sbt [hunk 2]
```diff
@@ -9,27 +9,36 @@ ThisBuild / publishFullName := "Ross A. Baker"
...
+  .settings(
+    libraryDependencies ++= Seq(
+      "org.typelevel" %%% "jawn-parser" % JawnVersion,
+      "co.fs2" %%% "fs2-core" % Fs2Version,
+      "org.typelevel" %%% "jawn-ast" % JawnVersion % Test,
+      "org.scalameta" %%% "munit" % MunitVersion % Test,
+      "org.typelevel" %%% "munit-cats-effect-3" % MunitCatsEffectVersion % Test
+    ),
+    versionIntroduced := Map(
+      "3.0.0-M1" -> "1.0.1",
```
Comment (mpilquist) [inline build.sbt:35]: Unrelated but this key can be dropped completely I believe.

**Review by mpilquist — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:23]: The full cross? Unfortunately yes, for the tests, because they read files.

File: build.sbt (hunk 2 above)
Comment (armanbilge) [reply build.sbt:35]: What was it for?

**Comment by armanbilge** [comment]
Thank you @mpilquist ❤️

File: build.sbt (hunk 2 above)
Comment (mpilquist) [reply build.sbt:35]: sbt-spiewak used (uses?) it to determine what mimaPreviousArtifacts should be set to for each crossScalaVersion. We definitely don't need the M1/RC entries as we no longer compile for those, and the fact that 3.0.0 isn't here tells me it's no longer necessary.

Comment (armanbilge) [reply build.sbt:35]: Gotcha. I can drop that.

**Pull request merged**